src/components/Table/hooks/useTable.tsx

```diff
@@ -164,6 +163,9 @@ const useTable = <TData,>({
 
   const hasRowDetails = data.some((row) => row.details) && Boolean(expanded);
 
+  /** Since tanstack's useTable doesn't detect array object mutations, we need to manually create new data array (new ref) to trigger a re-render */
```

Comment:
The question is, why are the data mutated? :eyes:

Reply:
tbh this comment was part of the previous implementation (where again it was not correctly worded), so its gonna be removed. 

_(the useMemo used on line 167 is required for tanstack's useTable hook to avoid infinite rerenders)_